modules/portal/portal-impl/src/main/java/com/enonic/xp/portal/impl/url/ApiUrlBaseUrlResolver.java

```diff
@@ -19,12 +18,80 @@ final class ApiUrlBaseUrlResolver
 {
     private final ContentService contentService;
 
-    private final ApiUrlGeneratorParams params;
+    private final Supplier<String> application;
```

Comment:
Can we instead require api descriptor?

Reply:
It will be done in https://github.com/enonic/xp/issues/11007